app/validators/answers/number_answer_validator.py

```diff
@@ -124,15 +124,21 @@ def validate_numeric_range(self, answer_ranges):
     def validate_referred_numeric_answer_decimals(self, answer_ranges):
         answer_values = answer_ranges[self.answer["id"]]
 
-        if answer_values["min_referred"] is not None:
+        if (
+            answer_values["min_referred"] is not None
+            and answer_values["min_referred"] in answer_ranges
+        ):
```

Comment:
We should do the same to ensure the answers referenced by calculated summary source is also checked. I'm referring to `test_invalid_range` test case.